include/quda_arch.h

```diff
@@ -0,0 +1,9 @@
+#pragma once
+#include <quda_define.h>
+
+#if defined(QUDA_TARGET_CUDA)
+#include <cuda.h>
+#include <cuda_runtime.h>
+#elif defined(QUDA_TARGET_HIP)
+#include <hip/hip_runtime.h>
+#endif
```

Comment:
New file to include runtime headers so they could be moved out of quda_internal.h sadly can't easily hide this in include/targets/ as tests seem to include headers which occasionally needed.